crocolaketools/converter/converter.py

```
@@ -810,6 +816,30 @@ def sort_rows(self,df):
 
         return df
 
+#------------------------------------------------------------------------------#
+## Convert units
+    def convert_units(self, ddf):
+        """Apply unit conversions defined in self.cols_to_convert.
+        
+        Arguments:
+        ddf -- dask dataframe
+        
+        Returns:
+        ddf -- updated dask dataframe with converted units
+        """
+
+        for col, conversion_key in self.cols_to_convert.items():
+            if conversion_key == "skip":
+                continue
+            if conversion_key not in unit_converter.conversion_map:
+                raise ValueError(f"No conversion defined for '{conversion_key}'")
+            if col not in ddf.columns: # column not found in df - skipping
```

I would raise an error here instead of letting the code silently succeed. I imagine that, for this `if` to be triggered, the user either is mispelling a column name or uses a column that is not actually there (e.g. DOXY but they're using the physical dataset) without realising it.